geniza/corpus/management/commands/tei_to_annotation.py

```diff
@@ -0,0 +1,347 @@
+            )
+            # FIXME: this is probably problematic for transcriptions currently
+            # split across two TEi files...
+            # maybe filter by date created also, so we only remove
+            # annotations created before the current script run?
+            if existing_annos:
+                print(
+                    "Removing %s existing annotation(s) for %s on %s "
+                    % (len(existing_annos), footnote.source, doc.manifest_uri)
+                )
+                for anno in existing_annos:
+                    annotation_client.delete(anno["id"])
+
+            for i, block in enumerate(html_blocks):
+                # if this is not the first block and the label suggests new image,
+                # increment canvas index
+                if i != 0 and tei.label_indicates_new_page(block["label"]):
+                    canvas_index += 1
+
+                # get the canvas uri for this section of text
+                annotation_target = iiif_canvases[canvas_index]
+
+                anno = new_transcription_annotation()
+                # link to digital edition footnote via source URI
+                anno["dc:source"] = footnote.source.uri
+
+                anno.target.source.id = annotation_target
+                anno.target.source.partOf.type = "Manifest"
+                anno.target.source.partOf.id = doc.manifest_uri
+
+                # apply to the full canvas using % notation
+                # (using nearly full canvas to make it easier to edit zones)
+                anno.target.selector.value = "xywh=percent:1,1,98,98"
+                # anno.selector.value = "%s#xywh=pixel:0,0,%s,%s" % (annotation_target, canvas.width, canvas.height)
+
+                # add html and optional label to annotation text body
+                # TODO: specify language in html if known
+                anno.body[0].value = tei.lines_to_html(block["lines"])
+                if block["label"]:
+                    anno.body[0].label = block["label"]
+
+                anno["schema:position"] = i + 1
+
+                # print(anno) # can print for debugging
+
+                created = annotation_client.create(anno)
+                if created:
+                    self.stats["created"] += 1
+
+        print(
+            "Processed %(xml)d TEI file(s). \nCreated %(created)d annotation(s)."
+            % self.stats
+        )
+
+    def get_footnote_editions(self, doc):
+        # extend to return digital edition or edition
+        # (digital edition if from previous run of this script)
+        return doc.footnotes.filter(
+            models.Q(doc_relation__contains=Footnote.EDITION)
+            | models.Q(doc_relation__contains=Footnote.DIGITAL_EDITION)
+        )
+
+    def migrate_footnote(self, footnote, document):
+        # convert existing edition footnote to digital edition
+        # OR make a new one if the existing footnote has other information
```

Comment:
How should we handle reconciling this with the automatic processes on `post_save` in the footnote branch? Is there something different that needs to happen here?

Reply:
Hmm, good point... I hadn't thought about the fact that those signals would be in place when this script runs (once we merge). My inclination is to disable those signal handles for the duration of this script. That seems simpler to me — then the signals are for annotation editing going forward. What do you think?

Reply:
Yes, I think disabling them makes sense. We'll probably need to modify it here slightly to account for the missing `content` field though!

Reply:
We can't remove `content` until after this release, since we need to use it to identify and migrate the correct edition footnotes. I'm planning to adjust your changes in the other branch to account for that, and we can plan to have a micro release immediately after the transcription migration release to clean up the content field once we're done with the migration.